Support for Evaporative cooling in thermo mode.

smaurer3 opened the pull request:
When I switched my system to Evaporative cooling - thermo mode, the plugin would crash with a traceback error indicate a Key error "fanSpeed" on line 230 of hub.py.  Inspecting the log revealed that when in thermo my climate uses a setpoint in the range of 1 - 8 for the fan speed.  The XML key for this is setPoint.

I have added support so that if the unit is in evap - thermo mode it will use the key value SetPoint instead of fanSpeed and will construct the correct send command with the mode as thermo and fanSpeed sent as set point.

I have tested this over the last few days on my unit and it has been working perfectly.  The HA climate integration still treats it is as changing the fan speed - this is similar to how the Bonaire Navigator controller works when in this mode.

I haven't thought of a better way to do this as the HA climate integrations isn't designed for a thermo mode that targets a fan speed like the bonaire units do.

boc-the-git commented:
Hey friends! I've recently jumped on to help bremor with the maintenance of the integration moving forward.

Most of the open issues are 2+ years old and obviously the information on them is a bit stale.
Can you please provide any update on this issue, giving as much detail as possible?

I'm keen to hear any detail about the necessity of this @smaurer3

smaurer3 commented:
Hello,

 

I’ll start with what my HVAC setup it:

I have a Bonaire Ducted Gas heater and a Bonaire Evaporative cooling system, both installed by the builder around 7 years ago.  I’d assume my setup is common as it is an inclusion on many project built homes.

 

When the Navigator controller is put in to cooling mode (evap), it controls the Evaporative cooler, and heating mode obviously controls the heater.

 

I have never encountered any issues with the integration while the system is in heating mode.

 

The evaporative cooler has 6 different modes of operation – “Thermo”, “Manual”, “Fan”, “Boost”, “Pad Clean” and “Pump Only”
“Pad Clean” and “Pump Only” are maintenance tasks and so far I can’t see anything in the debug log for the integration when I select them – it’s seems like the hardware is not reporting anything about these 2 modes.  It continues to report the actual cooling operation mode it was last in.

 

On the original Bremor version of this integration, if my evaporative cooler is operating in Thermo mode, which is the main mode we operate it in, the plugin crashes, I believe I explained a little in my pull request – the code in my fork fixes this by recognising the thermo mode and generating the correct commands to send for thermo.

When in Thermo mode the Setpoint defines the fan speed (it varies the fan speed depending on the temperature – hence been setpoint), in Boost mode the fan speed cannot be changed from 8.

 

The main issue with the bremor version still looks for fanSpeed when the system in is Thermo mode, but the system uses setpoint.

 

Currently my fork doesn’t report back in the entity state or attribute any way of knowing if it is in manual or thermo mode.

 

I can switch between these cooling “modes” using bonaire_myclimate.send_raw_command: examples – 

 

Change to Thermo Mode Fan Set Point 6

action: bonaire_myclimate.send_raw_command

data:

  raw_command:

    mode: thermo

    setPoint: 6

 

Changed to manual mode fan Speed 2

action: bonaire_myclimate.send_raw_command

data:

  raw_command:

    mode: manual

    fanSpeed: 2

 

Changed to Boost

action: bonaire_myclimate.send_raw_command

data:

  raw_command:

    mode: boost

 

Last mode when turned off “fan” – Fan Only, Start in Thermo:

action: bonaire_myclimate.send_raw_command

data:

  raw_command:

    system: on

    type: evap

    mode: thermo

 

 

I have included debug logs of me switching the Cooling mode to all 4 “Thermo”, “Manual”, “Fan”, “Boost” modes – I was controlling this from the Navigator Controller.  The error the is present in the Bremor version doesn’t appear in my logs as I am running my modified (forked) version of the integration.

 

Log Timestamp Key Events:

2024-10-27 07:42:55.221 – I turn the system on, it was already in thermos mode

2024-10-27 07:45:21.954 – Changed to Manual

2024-10-27 07:46:50.659  -Changed to Fan

2024-10-27 07:48:19.364 – Changed to Boost

2024-10-27 07:48:52.414 – Changed back to Thermo

 

 

2024-10-27 07:42:32.727 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:42:32.727 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>off</system><type>evap</type><zoneList>Common</zoneList><mode>thermo</mode><setPoint>4</setPoint><roomTemp id="1">20</roomTemp><roomTemp id="2">20</roomTemp><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:42:55.221 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:42:55.221 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>thermo</mode><setPoint>1</setPoint><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:44:00.174 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:44:00.175 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>thermo</mode><setPoint>1</setPoint><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:44:48.615 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending delete

2024-10-27 07:44:48.615 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><delete>connection</delete></myclimate>

2024-10-27 07:44:49.786 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server connection lost

2024-10-27 07:44:51.617 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending discovery

2024-10-27 07:44:51.617 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>discovery</get><ip>192.168.1.204</ip><platform>android</platform><version>1.0.0</version></myclimate>

2024-10-27 07:44:55.362 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Connected to the Bonaire MyClimate Wi-Fi device

2024-10-27 07:44:56.726 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:44:56.726 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>discovery</response><ip>192.168.1.105</ip><serial>502-0173-800043</serial><version>1.2.3</version><port>10002</port><status>router</status></myclimate>

2024-10-27 07:44:56.727 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending installation

2024-10-27 07:44:56.727 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>installation</get></myclimate>

2024-10-27 07:44:58.186 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:44:58.187 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>installation</response><appliance><type>heat</type><zoneList>1,2,Common</zoneList></appliance><appliance><type>cool</type><zoneList></zoneList></appliance><appliance><type>evap</type><zoneList>Common</zoneList></appliance><zoneName id="1">Zone 1</zoneName><zoneName id="2">Zone 2</zoneName><zoneName id="Common">Common</zoneName></myclimate>

2024-10-27 07:44:58.187 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending getzoneinfo

2024-10-27 07:44:58.188 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>getzoneinfo</get></myclimate>

2024-10-27 07:44:59.456 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:44:59.456 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>getzoneinfo</response><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>thermo</mode><setPoint>1</setPoint><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:45:21.954 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:45:21.954 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>manual</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:46:26.906 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:46:26.907 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>manual</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:46:28.170 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:46:28.170 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>manual</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">21</roomTemp></myclimate>

2024-10-27 07:46:50.659 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:46:50.659 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>fan</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">21</roomTemp></myclimate>

2024-10-27 07:47:34.381 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:47:34.382 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>fan</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">20</roomTemp></myclimate>

----------------------------------------------------------------------------------------------

2024-10-27 07:47:56.874 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:47:56.874 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>fan</mode><fanSpeed>2</fanSpeed><roomTemp id="Common">20</roomTemp></myclimate>

2024-10-27 07:48:19.364 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:48:19.364 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>boost</mode><fanSpeed>8</fanSpeed><roomTemp id="Common">20</roomTemp></myclimate>

-------------------------------------------------------------------------------------------

2024-10-27 07:48:41.621 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending delete

2024-10-27 07:48:41.621 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><delete>connection</delete></myclimate>

2024-10-27 07:48:41.855 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:48:41.856 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><post>postzoneinfo</post><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>boost</mode><fanSpeed>8</fanSpeed><roomTemp id="Common">21</roomTemp></myclimate>

2024-10-27 07:48:42.915 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server connection lost

2024-10-27 07:48:44.622 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending discovery

2024-10-27 07:48:44.622 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>discovery</get><ip>192.168.1.204</ip><platform>android</platform><version>1.0.0</version></myclimate>

2024-10-27 07:48:48.323 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Connected to the Bonaire MyClimate Wi-Fi device

2024-10-27 07:48:49.684 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:48:49.685 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>discovery</response><ip>192.168.1.105</ip><serial>502-0173-800043</serial><version>1.2.3</version><port>10002</port><status>router</status></myclimate>

2024-10-27 07:48:49.685 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending installation

2024-10-27 07:48:49.685 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>installation</get></myclimate>

2024-10-27 07:48:51.145 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:48:51.145 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>installation</response><appliance><type>heat</type><zoneList>1,2,Common</zoneList></appliance><appliance><type>cool</type><zoneList></zoneList></appliance><appliance><type>evap</type><zoneList>Common</zoneList></appliance><zoneName id="1">Zone 1</zoneName><zoneName id="2">Zone 2</zoneName><zoneName id="Common">Common</zoneName></myclimate>

2024-10-27 07:48:51.146 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending getzoneinfo

2024-10-27 07:48:51.146 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Sending: <myclimate><get>getzoneinfo</get></myclimate>

2024-10-27 07:48:52.414 INFO (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Server data received

2024-10-27 07:48:52.414 DEBUG (MainThread) [custom_components.bonaire_myclimate.BonairePyClimate.hub] Received: <myclimate><response>getzoneinfo</response><system>on</system><type>evap</type><zoneList>Common</zoneList><mode>thermo</mode><setPoint>1</setPoint><roomTemp id="Common">21</roomTemp></myclimate>

 

 

From: boc-the-git ***@***.*** ***@***.***> > 
Sent: Thursday, 26 September 2024 10:01 PM
To: bremor/bonaire_myclimate ***@***.*** ***@***.***> >
Cc: smaurer3 ***@***.*** ***@***.***> >; Mention ***@***.*** ***@***.***> >
Subject: Re: [bremor/bonaire_myclimate] Support for Evaporative cooling in thermo mode. (PR #30)

 

Hey friends! I've recently jumped on to help bremor with the maintenance of the integration moving forward.

Most of the open issues are 2+ years old and obviously the information on them is a bit stale.
Can you please provide any update on this issue, giving as much detail as possible?

I'm keen to hear any detail about the necessity of this @smaurer3 <https://github.com/smaurer3> 

—
Reply to this email directly, view it on GitHub <https://github.com/bremor/bonaire_myclimate/pull/30#issuecomment-2376739944> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AJF3DMWX5DXGMJJQPREGQHLZYPZOFAVCNFSM6AAAAABNMIPQCCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGNZWG4ZTSOJUGQ> .
You are receiving this because you were mentioned.Message ID: ***@***.*** ***@***.***> >

boc-the-git commented:
Apologies, I haven't had a chance to get into the weeds on this, and I don't yet have a read on whether this is useful to merge into master. Would love to hear any feedback from other people with similar setups.

MoFtZ commented:
> Apologies, I haven't had a chance to get into the weeds on this, and I don't yet have a read on whether this is useful to merge into master. Would love to hear any feedback from other people with similar setups.

Is there anyone for which `evap` with `thermo` actually does work?

I found this [old link](https://github.com/bremor/bonaire_myclimate/issues/19#issuecomment-716223969) which appears to have a `setPoint` command with `evap` and `thermo` option.

And [this link](https://github.com/bremor/bonaire_myclimate/issues/12#issuecomment-690863675) which is for `evap` but does not mention `thermo`.

In any case, I think this PR is insufficient. For example, we may need to consider [this code](https://github.com/bremor/bonaire_myclimate/blob/b4dbd42289a36f0908f513b0f15fdf3c402f5dca/custom_components/bonaire_myclimate/BonairePyClimate/hub.py#L258) (e.g. do something different when `thermo` is set) and [this code](https://github.com/bremor/bonaire_myclimate/blob/b4dbd42289a36f0908f513b0f15fdf3c402f5dca/custom_components/bonaire_myclimate/BonairePyClimate/hub.py#L294) (e.g. set `thermo` in the commands, or remember the last setting like [this code](https://github.com/bremor/bonaire_myclimate/blob/b4dbd42289a36f0908f513b0f15fdf3c402f5dca/custom_components/bonaire_myclimate/BonairePyClimate/hub.py#L31)).
